### PR TITLE
ddl: use latest PD address to register lightning

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/48680

Problem Summary:

### What is changed and how it works?

as title

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/index.go [hunk 1]
```diff
@@ -770,10 +770,11 @@ func pickBackfillType(ctx context.Context, job *model.Job, unique bool, d *ddlCt
 			if err != nil {
 				return model.ReorgTypeNone, err
 			}
+			pdLeaderAddr := d.store.(tikv.Storage).GetRegionCache().PDClient().GetLeaderAddr()
 			if variable.EnableDistTask.Load() {
-				_, err = ingest.LitBackCtxMgr.Register(ctx, unique, job.ID, d.etcdCli, job.ReorgMeta.ResourceGroupName)
+				_, err = ingest.LitBackCtxMgr.Register(ctx, unique, job.ID, d.etcdCli, pdLeaderAddr, job.ReorgMeta.ResourceGroupName)
 			} else {
-				_, err = ingest.LitBackCtxMgr.Register(ctx, unique, job.ID, nil, job.ReorgMeta.ResourceGroupName)
+				_, err = ingest.LitBackCtxMgr.Register(ctx, unique, job.ID, nil, pdLeaderAddr, job.ReorgMeta.ResourceGroupName)
```
Comment (ywqzzy) [inline pkg/ddl/index.go:777]: Will be removed in https://github.com/pingcap/tidb/pull/48645

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: pkg/ddl/index.go (hunk 1 above)
Comment (Benjamin2037) [reply pkg/ddl/index.go:777]: 48645 is not block 7.5， we can merge it later.

**Review by tangenta — APPROVED** [review]

**Pull request merged**